src/data/mystery-encounters/mystery-encounters.ts

```diff
@@ -3,14 +3,124 @@ import {DarkDealEncounter} from "./dark-deal";
 import {MysteriousChallengersEncounter} from "./mysterious-challengers";
 import {MysteriousChestEncounter} from "./mysterious-chest";
 import {FightOrFlightEncounter} from "#app/data/mystery-encounters/fight-or-flight";
+import { Biome } from "../enums/biome";
 
-export const allMysteryEncounters: MysteryEncounter[] = [];
 
+// Only add your MysterEncounter here if you want it to be in every biome.
+// We recommend designing biome-specific encounters for better flavor and variance
 export function initMysteryEncounters() {
-  allMysteryEncounters.push(
-    MysteriousChallengersEncounter,
-    MysteriousChestEncounter,
-    DarkDealEncounter,
-    FightOrFlightEncounter
-  );
+  for (const biome of mysteryEncountersByBiome.keys()) {
+    mysteryEncountersByBiome.get(biome).push(
+      MysteriousChallengersEncounter,
+      MysteriousChestEncounter,
+      DarkDealEncounter,
+      FightOrFlightEncounter
+    );
+  }
 }
+
+
+// Add your MysteryEncounter to a biome to enable it to show up in that biome.
+export const mysteryEncountersByBiome = new Map<Biome, MysteryEncounter[]>([
```

Comment:
but really, most encounters should live in here. doubling up on biomes is encouraged.